### DISCUSSION TITLE
Help with php artisan key:generate not showing app key

### DISCUSSION THREAD

**Discussion by gurds91** [laravel/framework#49010] [Q&A] [answered]
Hi guys, I have good knowledge of the fundamentals of web stack - html, css, js, php. Currently I am trying to learn and understand laravel framework in php.

I started a new laravel project and am configuring the .env file. I ran command php artisan key:generate, it says key has been generated but it is not showing next to APP_KEY in the .env file.

Any help please, I tried chat gpt but couldn't solve this.

**Comment by antimech** [comment]
I never had this problem, even tried this now through `composer create-project ...` which I haven't used for a long time - works just as fine as the Laravel Installer (actually even better than it used to be, not as good as the installer though). Tell us how exactly we can reproduce it.
For example, have you either used the installer (`laravel new example-app`) or `composer create-project laravel/laravel example-app`?
Which PHP version do you use?

**Comment by gurds91** [comment]
I used composer create-project.. I have php version 8.2.

Is annoying why the app key is not showing in the .env file. Is there any way to find out what the app key is so I can type it in the .env file next to APP_KEY?

**Reply by antimech** [reply]
Run this:
```Shell
php artisan key:generate --show
```
and copy-paste it.

In case it won't help just use this, don't use it in your production server though:
```env
APP_KEY=base64:cIAZm9N4b/jlWLNm55/18PbcKN9lny36KiE0t3Ox4YI=
```

**Reply by antimech** [reply]
Just fired up:
```Shell
composer create-project laravel/laravel example-app
```
with PHP 8.2 on my Linux system and it generated the `APP_KEY` automatically, like it should be. Seems like an issue on your end. What is your OS? And could you paste the exact content of your `.env` (just in case you missed something)?

**Reply by gurds91** [reply]
APP_NAME=My Laravel App
APP_ENV=local
APP_KEY=
APP_DEBUG=true
APP_URL=http://localhost

LOG_CHANNEL=stack
LOG_DEPRECATIONS_CHANNEL=null
LOG_LEVEL=debug

DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=laravel
DB_USERNAME=root
DB_PASSWORD= (I didn't share password here)

BROADCAST_DRIVER=log
CACHE_DRIVER=file
FILESYSTEM_DISK=local
QUEUE_CONNECTION=sync
SESSION_DRIVER=file
SESSION_LIFETIME=120

MEMCACHED_HOST=127.0.0.1

REDIS_HOST=127.0.0.1
REDIS_PASSWORD=null
REDIS_PORT=6379

MAIL_MAILER=smtp
MAIL_HOST=mailpit
MAIL_PORT=1025
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="hello@example.com"
MAIL_FROM_NAME="${APP_NAME}"

AWS_ACCESS_KEY_ID=
AWS_SECRET_ACCESS_KEY=
AWS_DEFAULT_REGION=us-east-1
AWS_BUCKET=
AWS_USE_PATH_STYLE_ENDPOINT=false

PUSHER_APP_ID=
PUSHER_APP_KEY=
PUSHER_APP_SECRET=
PUSHER_HOST=
PUSHER_PORT=443
PUSHER_SCHEME=https
PUSHER_APP_CLUSTER=mt1

VITE_APP_NAME="${APP_NAME}"
VITE_PUSHER_APP_KEY="${PUSHER_APP_KEY}"
VITE_PUSHER_HOST="${PUSHER_HOST}"
VITE_PUSHER_PORT="${PUSHER_PORT}"
VITE_PUSHER_SCHEME="${PUSHER_SCHEME}"
VITE_PUSHER_APP_CLUSTER="${PUSHER_APP_CLUSTER}"


my OS is windows 10.

**Reply by antimech** [reply]
Your file is fine, I thought it was a parsing issue.
Windows could definitely be the reason. Try installing [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10) and create a project again. If this doesn't help then install [Docker](https://laravel.com/docs/10.x#sail-on-windows).

**Comment by gurds91** [comment, marked as answer]
hi somehow I managed to solve this, maybe by luck.

I downloaded open SSL which is open source and ran this command in cmd          openssl rand -base64 32

this effectively runs and spits out an app key that you can use. I then copied and pasted this app key into the .env file in my laravel project. I then went and saved it but I accidentally saved the .env file as env.txt so this created another file in my project. After deleting this plain text env file, I noticed that the normal app key generated by php artisan magically appeared in my original .env file ;)

Pretty good, maybe a solution others can use to fix this issue if they run into problems with this.

**Reply by antimech** [reply]
I just fired up a clean VM with:
* Windows 10 22H2
* XAMPP 8.2.4 (PHP 8.2.4)
* Composer 2.6.5

And it works just as expected.

**Reply by antimech** [reply]
Please mark your comment as the answer, it helps others find the solution
![](https://github.com/laravel/framework/assets/13885653/0d4d8bd8-a827-48f6-a69e-4bb4db1959b9)

And close this discussion as resolved
![](https://github.com/laravel/framework/assets/13885653/20ec0cea-75fc-496b-8799-70f9b6bbd78a)

**Comment by fauzan05** [comment]
I have an other solution for that, I used "composer install" or "composer update" first, then you can type "php artisan key:generate" as well and the key will be generate automatically.
